diff --git a/CMakeLists.txt b/CMakeLists.txt
index d9ad03c..9f535c0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.17)
-project(QIO VERSION 2.5.1 LANGUAGES C)
+project(QIO VERSION 3.0.0 LANGUAGES C)
 
 include(GNUInstallDirs)
 include(CheckIncludeFile)
@@ -18,6 +18,13 @@ set(QIO_DML_BUF_BYTES "262144"  CACHE STRING "Maximum DML Buffer Size in bytes")
 set(QMP_DIR "" CACHE STRING "QMP Install Directory")
 set(CLime_DIR "" CACHE STRING "C-Lime library DIrectory")
 
+unset(QMP_CFLAGS CACHE)
+set(QMP_CFLAGS "")
+unset(QMP_LDFLAGS CACHE)
+set(QMP_LDFLAGS "")
+unset(QMP_LIBS CACHE)
+set(QMP_LIBS "")
+
 if(QIO_ENABLE_PARALLEL_BUILD)
   message(STATUS "Setting Parallel arch to parscalar")
   set(PARALLEL_ARCH "parscalar")
@@ -31,6 +38,44 @@ else()
   set(ARCHDEF_SUBSTITUTION "ARCH_SCALAR")
 endif()
 
+if(QIO_ENABLE_PARALLEL_BUILD AND TARGET QMP::qmp)
+  get_target_property(QMP_INCLUDE_DIRS QMP::qmp INTERFACE_INCLUDE_DIRECTORIES)
+  if(QMP_INCLUDE_DIRS)
+    foreach(dir ${QMP_INCLUDE_DIRS})
+      set(QMP_CFLAGS "${QMP_CFLAGS} -I${dir}")
+    endforeach()
+  else()
+    message(WARNING "TARGET QMP::qmp has no property INTERFACE_INCLUDE_DIRECTORIES")
+  endif()
+  get_target_property(QMP_LIBRARIES QMP::qmp IMPORTED_LOCATION)
+  if(NOT QMP_LIBRARIES OR QMP_LIBRARIES STREQUAL "NOTFOUND")
+    message(WARNING "TARGET QMP::qmp has no property IMPORTED_LOCATION valid; Trying locating QMP libs via INTERFACE_LINK_LIBRARIES")
+    get_target_property(QMP_LIBS_LINK QMP::qmp INTERFACE_LINK_LIBRARIES)
+    if (QMP_LIBS_LINK)
+      foreach(link ${QMP_LIBS_LINK})
+        if(IS_ABSOLUTE "${link}" AND EXISTS "${link}")
+          set(QMP_LIBRARIES "${link}")
+          break()
+        endif()
+      endforeach()
+    else()
+      message(WARNING "TARGET QMP::qmp has no property IMPORTED_LOCATION valid, nor INTERFACE_LINK_LIBRARIES")
+    endif()
+  endif()
+  if(QMP_LIBRARIES AND EXISTS ${QMP_LIBRARIES})
+    get_filename_component(QMP_LIB_DIR "${QMP_LIBRARIES}" DIRECTORY)
+  else()
+    list(GET QMP_INCLUDE_DIRS 0 QMP_INCLUDE_DIR)
+    get_filename_component(QMP_INCLUDE_PATH "${QMP_INCLUDE_DIR}" PATH)
+    set(QMP_LIB_DIR "${QMP_INCLUDE_PATH}/lib")
+    message(WARNING "Could not locate QMP libs; Using fallback library path: ${QMP_LIB_DIR}")
+    if(NOT EXISTS "${QMP_LIB_DIR}/libqmp.a" AND NOT EXISTS "${QMP_LIB_DIR}/libqmp.so")
+      message(WARNING "Fail to find file: ${QMP_LIB_DIR}/libqmp.*")
+    endif()
+  endif()
+  set(QMP_LDFLAGS "-L${QMP_LIB_DIR} -Wl,-rpath=${QMP_LIB_DIR}")
+  set(QMP_LIBS "-lqmp")
+endif()
    
 if( QIO_ENABLE_PARALLEL_IO )
   message(STATUS "Enabling Paralle I/O")
@@ -78,6 +123,8 @@ check_type_size("unsigned long" SIZEOF_UNSIGNED_LONG)
 check_type_size("unsigned long long" SIZEOF_UNSIGNED_LONG_LONG)
 check_type_size("unsigned short" SIZEOF_UNSIGNED_SHORT)
 
+get_filename_component(QIO_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX} ABSOLUTE)
+set(QIO_PREFIX "${QIO_INSTALL_PREFIX}")
 configure_file(include/qio_config_internal.cmake.h.in include/qio_config_internal.h)
 
 # Deal with Sanitizer
diff --git a/bin/qio-config.cmake.in b/bin/qio-config.cmake.in
index ad7e9fe..2e5b136 100644
--- a/bin/qio-config.cmake.in
+++ b/bin/qio-config.cmake.in
@@ -9,20 +9,20 @@
 #
 # copied from qmp-config
 
-prefix=$(dirname "$0")/..
+prefix="@QIO_PREFIX@"
 version="@PROJECT_VERSION@"
 
 qio_parallel_arch="@PARALLEL_ARCH@"
 qio_cc="@CMAKE_C_COMPILER@"
-qio_cflags="-I$prefix/include"
-qio_ldflags="-L$prefix/lib -Wl,-rpath=$prefix/lib"
+qio_cflags="-I$prefix/include @QMP_CFLAGS@"
+qio_ldflags="-L$prefix/lib -Wl,-rpath=$prefix/lib @QMP_LDFLAGS@"
 if test "X@CLime_LIBDIR@" != "X";
 then
   qio_cflags=$qio_cflags" -I@CLime_INCDIR@"
-  qio_ldflags=$qio_ldflags" -L@CLime_LIBDIR@ -Wl.-rpath=@CLime_LIBDIR@"
+  qio_ldflags=$qio_ldflags" -L@CLime_LIBDIR@ -Wl,-rpath=@CLime_LIBDIR@"
 fi
  
-qio_libs="-lqio -llime"
+qio_libs="-lqio -llime @QMP_LIBS@"
 qio_ranlib="@CMAKE_RANLIB@"
 qio_ar="@CMAKE_AR@"
 
